### PR TITLE
Add SSL support for heka_tcp.lua input and output sandboxes

### PR DESCRIPTION
**Pull request by bbinet**
Fixes https://github.com/trink/hindsight/issues/27

### CODE REVIEW

**Comment by trink** [comment]
Can you bump the version to 0.16.0 here https://github.com/mozilla-services/lua_sandbox/blob/master/CMakeLists.txt#L9.  When I land some Hindsight changes later today I will update the required sandbox version there too.

**Comment by bbinet** [comment]
sure: 627f54f38ec08206f8a8b331a4fe49cf8c828a55

File: sandboxes/heka/input/heka_tcp.lua [hunk 1]
```diff
@@ -63,9 +73,18 @@ function process_message()
...
                     local client = s:accept()
                     if client then
+                        local caddr, cport = client:getpeername()
+                        if not caddr then
+                            caddr = "unknown"
+                            cport = 0
+                        end
+                        if ssl_ctx then
+                            client = ssl.wrap(client, ssl_ctx)
+                            assert(client:dohandshake())
```
Comment (trink) [inline sandboxes/heka/input/heka_tcp.lua]: This will terminate the plugin on a timeout correct? (probably not what we want)

Comment (bbinet) [inline sandboxes/heka/input/heka_tcp.lua]: I've removed the assert statement and rebased my branch.

**Comment by trink** [comment]
Do you want me to add the conditional compilation? as per: https://github.com/trink/hindsight/issues/27#issuecomment-193896012 or do you want to take a crack at it?

**Comment by bbinet** [comment]
I'd be happy if you can add the conditional compilation, as I'm not sure how to do it.

**Pull request merged**